### ISSUE TITLE
deployProxy prints errors to the console and reports deployment failed even though the deployment succeeded

### ISSUE THREAD

**Issue by mverzilli** [OpenZeppelin/openzeppelin-upgrades#135]
I succeeded to deploy a proxy and its implementation but the script is throwing. You can verify that the contract was actually deployed and the contract architecture is correct: https://rinkeby.etherscan.io/address/0xb4463a644DD6E16ccF518df5b8bf0fB9b36ef68c 

```
All contracts have already been compiled, skipping compilation.
(node:33906) UnhandledPromiseRejectionWarning: Error: contract not deployed (contractAddress="0xb4463a644DD6E16ccF518df5b8bf0fB9b36ef68c", operation="getDeployed", code=UNSUPPORTED_OPERATION, version=contracts/5.0.2)
    at Logger.makeError ((...)/node_modules/@ethersproject/logger/src.ts/index.ts:205:28)
    at Logger.throwError ((...)/node_modules/@ethersproject/logger/src.ts/index.ts:217:20)
    at (...)/node_modules/@ethersproject/contracts/src.ts/index.ts:770:32
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at main ((...)/scripts/create-my-contract.js:7:3)
(node:33906) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:33906) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The script I ran is based on the quickstart example:

```
// scripts/create-my-contract.js
const { ethers, upgrades } = require("@nomiclabs/buidler");

async function main() {
  const MyContract = await ethers.getContractFactory("MyContract");
  const myContract = await upgrades.deployProxy(MyContract, [42]);
  await myContract.deployed();
  console.log("MyContract deployed to:", myContract.address);
}

main().then(() => console.log('Done!'));
```

The contract code is:

```
// contracts/MyContract.sol
// SPDX-License-Identifier: MIT
pragma solidity ^0.6.0;

pragma solidity >=0.4.24 <0.7.0;


/**
 * @title Initializable
 *
 * @dev Helper contract to support initializer functions. To use it, replace
 * the constructor with a function that has the `initializer` modifier.
 * WARNING: Unlike constructors, initializer functions must be manually
 * invoked. This applies both to deploying an Initializable contract, as well
 * as extending an Initializable contract via inheritance.
 * WARNING: When used with inheritance, manual care must be taken to not invoke
 * a parent initializer twice, or ensure that all initializers are idempotent,
 * because this is not dealt with automatically as with constructors.
 */
contract Initializable {

  /**
   * @dev Indicates that the contract has been initialized.
   */
  bool private initialized;

  /**
   * @dev Indicates that the contract is in the process of being initialized.
   */
  bool private initializing;

  /**
   * @dev Modifier to use in the initializer function of a contract.
   */
  modifier initializer() {
    require(initializing || isConstructor() || !initialized, "Contract instance has already been initialized");

    bool isTopLevelCall = !initializing;
    if (isTopLevelCall) {
      initializing = true;
      initialized = true;
    }

    _;

    if (isTopLevelCall) {
      initializing = false;
    }
  }

  /// @dev Returns true if and only if the function is running in the constructor
  function isConstructor() private view returns (bool) {
    // extcodesize checks the size of the code stored in an address, and
    // address returns the current address. Since the code is still not
    // deployed when running a constructor, any checks on its code size will
    // yield zero, making it an effective way to detect if a contract is
    // under construction or not.
    address self = address(this);
    uint256 cs;
    assembly { cs := extcodesize(self) }
    return cs == 0;
  }

  // Reserved storage space to allow for layout changes in the future.
  uint256[50] private ______gap;
}

contract MyContract is Initializable {
    uint256 public x;

    function initialize(uint256 _x) public initializer {
        x = _x;
    }

    function talk() public view returns (uint256) {
      return x;
    }
}
```

**Comment by frangio** [comment]
This may be related to #122. `myContract.deployed()` was not working, it was fixed in `master` but I've only published it now. Can you try again and let me know if it works?

**Comment by frangio** [comment]
Fixed in #129 and released in 1.0.1.
